### ISSUE TITLE
The sample is not exhibiting SSO feature as it is prompting user to enter password even when user already logged in with authenticator app

### ISSUE THREAD

**Issue by pramodkumarhk** [Azure-Samples/ms-identity-android-java#64]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>Download this sample and build using android studio.
>Install Authenticator app in the Nexus 5 simulator (API 30) where we will be running this sample. 
>Open authenticator app and login with the account so that, other apps installed on this device can make use of SSO feature. 
>Now, I'll run the sample in the simulator. It will show the signin button, upon clicking on it, it will prompt me to choose the account from broker(authenticator app). However, after choosing the account, I dont expect it to prompt for credential again as I have already logged in with the account inside broker. 

### Any log messages given by the failure
> Incident ID uploaded - PTZHK92Y. + Screen sharing video are uploaded in zip file. 

### Expected/desired behavior
> I dont expect the app to prompt credential as I already logged in in authenticator app. I have also noticed that, if I sign out and then again signin then again it will ask for the password

### OS and Version?
> Nexus 5

### Versions
>> Nexus 5 API 30 simulator 

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.
[MSALAndroidSamleWithBrokerBroker.zip](https://github.com/Azure-Samples/ms-identity-android-java/files/9493780/MSALAndroidSamleWithBrokerBroker.zip)

**Comment by iamgusain** [comment]
HI @pramodkumarhk to enable SSO with the sample app you would need perform device registration from Authenticator app first.
You can register the device from "Authenticator->Settings->Device registration" page and signing in using the user credentials.
![image](https://user-images.githubusercontent.com/75644120/189241209-bc4d8448-f312-4ac2-9f0c-a8db2d28bec7.png)

**Comment by pramodkumarhk** [comment]
@iamgusain : Yes, after device registration SSO seem to work now. However, few points that needs clarification.
1. After I registered the device via authenticator app, sample app still prompted for the first time, I'm not sure why. After I logged in by entering the password, then it didnt ask the password again. I tried uninstall/install of the app, but it didnt prompt for the password. So, it was only once after device registration app prompted for password. Is it expected ?
2. I'm not convinced by the fact that, device registration process is mandatory to achieve SSO. I can achieve SSO using system browser without a broker and without device registration step. So, device registration becomes a mandatory step only if broker present on the device. Correct ?

**Comment by mohitc1** [comment]
Hi @pramodkumarhk 
For 1) It seems sample still asked password because, the app was not consented for this tenant/domain. Could you confirm if you saw consent screen? This should happen once only, and after that you should not be prompted for password for the scenario.
For 2) The broker does not work with system browser hence cannot provide SSO using it. Device registration enables primary refresh token protocol which then enables the SSO device wide.
Certain Microsoft applications can still do SSO with each other via broker, without device being registered, but that's not available widely and device wide yet.

**Comment by pramodkumarhk** [comment]
@iamgusain  @mohitc1 

1) There was no consent screen involved in the scenario, as I had already consented it well before trying out the device registration scenario. This is not a priority for me now as I cant reproduce it on the same device. I tried , unregistering/registering the device, uninstalling/installing the apps and broker and cant reproduce the issue on the device. Perhaps, it was just one time !!!
2) I understand the fact that, brokers use their own private cache and doesn't share the system browser cache. My point was that, if I do not want to register by device and still want to achieve SSO, then I can simply use system browser approach and achieve my requirement. Isn't it ? Of course, I might lose the features that broker offers which I'm okay with. I feel, Broker is making the device registration a mandatory step which is not really necessary. (There might be technical reasoning due to how the broker code is implemented as you mentioned about PRT coming into the picture, but purely from logical and functionality standpoint, I feel this is an additional step which could have been avoided) . Do you agree ?

**Comment by pramodkumarhk** [comment]
3.  After registering the device, though sample is exhibiting SSO feature. Teams app doesn't seem to pick this SSO account . it is still asking for credential. Refer the video in the zip file. I have also copied the authenticator logs and uploaded the incident ID **WQBNG9MB**
[WQBNG9MB_With_ScreenCapture.zip](https://github.com/Azure-Samples/ms-identity-android-java/files/9546473/WQBNG9MB_With_ScreenCapture.zip)

**Comment by mohitc1** [comment]
@pramodkumarhk Teams explicitly asks for login the first time by communicating that to broker. This is also suggested from logs.
